appserver/connectors/connectors-connector/src/main/java/com/sun/enterprise/connectors/connector/module/RarDetector.java

```diff
@@ -69,8 +69,24 @@
 @Service(name = RarDetector.ARCHIVE_TYPE)
 @Singleton
 public class RarDetector implements ArchiveDetector {
+    
+    /**
+     * Connector annotations for JCA 1.7
+     */
     private static final Class[] connectorAnnotations = new Class[]{
-            javax.resource.spi.Connector.class};
+            javax.resource.spi.Connector.class,
+            javax.resource.spi.ConfigProperty.class,
+            javax.resource.spi.ConnectionDefinition.class,
```

Comment:
I don't think this is correct as other applications can have these annotations and aren't rar files.

You need to see which ones are rar only and which ones can be used by clients of a rar